Classes/Utility/Store.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Store.swift
+//  Freetime
+//
+//  Created by Hesham Salman on 11/5/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+protocol Store: AnyObject {
+    associatedtype Model: Codable, Equatable
+
+    var key: String { get }
+    var defaults: UserDefaults { get }
+
+    var encoder: JSONEncoder { get }
+    var decoder: JSONDecoder { get }
+
+    var values: [Model] { get set }
```

Comment:
Really weary about a public `var` here.

1. No thread safety
2. Anyone can do anything

Also we probably need to revisit this so that mutations and lookups are O(1)